assets/wezterm-nautilus.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (C) 2022 Sebastian Wiesner <sebastian@swsnr.de>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+from gi import require_version
+require_version('Nautilus', '3.0')
+from gi.repository import Nautilus, GObject, Gio, GLib
+
+
+class OpenInWezTermAction(GObject.GObject, Nautilus.MenuProvider):
+    def __init__(self):
+        super().__init__()
+        session = Gio.bus_get_sync(Gio.BusType.SESSION, None)
+        self._systemd = Gio.DBusProxy.new_sync(session, Gio.DBusProxyFlags.NONE,
```

Comment:
I think it would be good to not require systemd here.  Assuming that this proxy creation fails if systemd is not present, then I'd suggest wrapping this in a `try`/`catch`, then in `_open_terminal` below use `subprocess` to spawn the terminal if `self_systemd is None`.

Would be good to verify that the subprocess branch of the code actually works! :-)

Reply:
I've added a check for systemd before using the systemd API.